yinsolidated/plugin/plugin.py

```diff
@@ -288,7 +285,45 @@ def _make_extension_element(statement, parent_elem):
     nsmap = {prefix: namespace}
 
     extension_element = etree.SubElement(parent_elem, tag, nsmap=nsmap)
-    extension_element.text = statement.arg
+
+    if _is_complex_extension(statement.i_extension):
+        argument_name, is_arg_yin_element = _get_exension_argument_details(
+            statement.i_extension
+        )
+        _add_statement_argument(
+            argument_name,
+            statement.arg,
+            namespace,
+            is_arg_yin_element,
+            extension_element,
+        )
+        _append_children(statement, extension_element)
+    else:
+        extension_element.text = statement.arg
+
+
+def _is_complex_extension(extension_statement):
+    extension_description = extension_statement.search_one('description')
+    return (
+        extension_description is not None and
+        "#yinformat" in extension_description.arg
```

Comment:
I'm not super crazy about using the description for this (especially for "public" extensions) but I can't think of anything better at the moment, so... I guess it's fine.

Reply:
alternatively we could use the `reference` field, where referencing RFC 6020 or some such would be the trigger.

Reply:
Ironically, that RFC is _not_ a good reference for the extension substatements. From §7.17.
> The substatements of an extension are defined by the extension, using some mechanism outside the scope of this specification.  Syntactically, the substatements MUST be YANG statements, or also defined using "extension" statements.

Reply:
Using the `reference` field just doesn't seem intuitive to me. I think it should be obvious when you look at the model that the extension has been tagged with some metadata that's being used programmatically. Just sticking "RFC XXXX" in a reference doesn't make that apparent at all. I also don't think putting the `#yinformat` tag in the reference statement makes much sense. The description is more of a free-form string, so to me, it fits better there. Plus extension descriptions don't end up in being presented on any of our UIs, so it seemed relatively harmless.